spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java

```diff
@@ -69,9 +70,24 @@ public DataType struct(Types.StructType struct, List<DataType> fieldResults) {
       if (field.doc() != null) {
         sparkField = sparkField.withComment(field.doc());
       }
+      if (field.writeDefault() != null) {
+        // Convert Iceberg default value to Spark SQL string representation. Spark stores default
+        // values as SQL strings in column metadata. The .sql() method formats literals correctly
+        // for each type
+        Object writeDefault = SparkUtil.internalToSpark(field.type(), field.writeDefault());
+        sparkField =
+            sparkField.withCurrentDefaultValue(Literal$.MODULE$.create(writeDefault, type).sql());
+      }
+      if (field.initialDefault() != null) {
+        // Same conversion for existence default values, used for existing rows when column is added
+        // to schema
```

Comment:
I'd just remove completely or say "Same conversion logic as write defaults but for initial defaults". No need to specify _when_ initial defaults (or in spark's terms, existence) is used in the code comment.